docs/langref/relay/intro.rst

```diff
@@ -0,0 +1,182 @@
+the computation graph-based intermediate representations currently employed by deep
+learning frameworks and compilers.
+
+The deep learning community has organically evolved a representation of machine learning
+models that was useful for the form of computation originally desired, i.e., a directed 
+acyclic graph of primitive functions. Computation graphs also serve a dual purpose as both a 
+compile-time and run-time data structure. While computation graphs are a good fit
+for models with a static topology, they make it tricky to incorporate control flow and
+abstraction. Additionally, the conflation of the description of a computation, the
+representation used for optimizing it, and the data structure used to execute it unnecessarily 
+hamper the goals of machine learning frameworks. We believe having a high-level, expressive language 
+designed for writing compiler optimizations is essential to the future of an end-to-end
+deep learning compiler stack.
+
+Relay's design is influenced by the authors' experience building advanced optimizing compilers
+for high-level languages, as well as by challenges presented by the current version of the
+TVM stack, and NNVM's IR. We address a few important challenges with Relay's design.
+Relay is an IR with closures, control-flow, recursion, and an advanced type system supporting
+complex shape relationships and symbolic dimensions. We define a series of
+automatic differentiation rules over the language, with the goal of enabling higher-order
+differentiation of programs with control flow and closures.
+
+==================
+Language
+==================
+
+Relay is a purely functional, differentiable intermediate representation.
+
+==================
+IR Reference
+==================
+
+The IR has a global environment that stores the set of definitions,
+constants, options, and attributes, and provides access to features
+including type inferecnce, constant evaluation, and more.
+
+~~~~~~~~~~
+Relay Node
+~~~~~~~~~~
+
+The fundamental unit of the IR is the node, which only contains a Span.
+
+.. code-block:: python
+
+    class Node:
+        span: Span
+
+==================
+Variables
+==================
+
+Relay has two notions of variables: local and global.
+Our design draws inspiration from LLVM, which differentiates between identifier types.
+This enables writers of optimizations to know precisely what an identifier references without needing
+information beyond the kind of identifier.
+
+Globals are written with `@`, locals are written with `%`, and variables written without a
+sigil name the corresponding operator. The distinction between global and local identifiers
+makes certain kinds of transformation easier. For example, inlining a global definition
+requires no analysis, as you can write a pass that directly inlines the definitions.
+Ensuring there is no spooky action at a distance; introducing a new identifier return
```

Comment:
Was this sentence clobbered during a merge or rebase? It no longer makes any sense and I didn't know what to change it to